test/com/facebook/buck/rules/macros/AbstractLocationMacroTest.java

```diff
@@ -0,0 +1,35 @@
+package com.facebook.buck.rules.macros;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.fail;
+
+import com.facebook.buck.core.exceptions.HumanReadableException;
+import java.util.Optional;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+public class AbstractLocationMacroTest {
+  @Test
+  public void testSplitSupplementaryOutput() {
+    assertThat(
+        AbstractLocationMacro.splitSupplementaryOutputPart("target[sup]").target,
+        Matchers.equalTo("target"));
+    assertThat(
+        AbstractLocationMacro.splitSupplementaryOutputPart("target[sup]").supplementaryOutput,
+        Matchers.equalTo(Optional.of("sup")));
+    assertThat(
+        AbstractLocationMacro.splitSupplementaryOutputPart("target").target,
+        Matchers.equalTo("target"));
+    assertThat(
+        AbstractLocationMacro.splitSupplementaryOutputPart("target").supplementaryOutput,
+        Matchers.equalTo(Optional.empty()));
+    try {
```

Comment:
Could you move the test case with the exception to a separate method? Other test cases are homogeneous and can stay in one method, but the the test case with an error stands out.

Reply:
Makes sense. Done.